ios/CameraView+AVCaptureSession.swift

```diff
@@ -125,16 +127,44 @@ extension CameraView {
       captureSession.removeOutput(videoOutput)
       self.videoOutput = nil
     }
+    if let depthOutput = depthOutput {
+      captureSession.removeOutput(depthOutput)
+      self.depthOutput = nil
+    }
     if video?.boolValue == true || enableFrameProcessor {
-      ReactLogger.log(level: .info, message: "Adding Video Data output...")
       videoOutput = AVCaptureVideoDataOutput()
       guard captureSession.canAddOutput(videoOutput!) else {
         invokeOnError(.parameter(.unsupportedOutput(outputDescriptor: "video-output")))
         return
       }
-      videoOutput!.setSampleBufferDelegate(self, queue: videoQueue)
-      videoOutput!.alwaysDiscardsLateVideoFrames = false
-      captureSession.addOutput(videoOutput!)
+      // TODO: Replace with `depth` prop value
+      if enableDepthData?.boolValue != true || videoDevice.activeDepthDataFormat == nil {
+        ReactLogger.log(level: .info, message: "Adding Video Data output...")
+        videoOutput!.setSampleBufferDelegate(self, queue: videoQueue)
+        videoOutput!.alwaysDiscardsLateVideoFrames = false
+        captureSession.addOutput(videoOutput!)
+      } else {
+        ReactLogger.log(level: .info, message: "Adding Video + Depth Data output...")
+        captureSession.addOutput(videoOutput!)
+        depthOutput = AVCaptureDepthDataOutput()
+        // Add a depth data output
+        if captureSession.canAddOutput(depthOutput!) {
+          captureSession.addOutput(depthOutput!)
+          depthOutput!.isFilteringEnabled = true
+          if let connection = depthOutput!.connection(with: .depthData) {
+            connection.isEnabled = true
+          } else {
+            print("No AVCaptureConnection")
+          }
+        } else {
+          invokeOnError(.parameter(.unsupportedOutput(outputDescriptor: "depth-output")))
+          captureSession.commitConfiguration()
```

Comment:
That should probably be removed

Reply:
Why? Don't we need to return early and throw an error if the use has requested depth data and the depth output can't be attached to the capture session?

Reply:
Yeah but the commit should be removed. I am committing the changes above anyways (`defer` block)